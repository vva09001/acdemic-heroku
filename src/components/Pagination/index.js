import React, { useCallback, useState, useEffect } from "react";

import { ButtonClick } from "../../theme/Button";

const Pagination = ({ maxPage, page, onPressEnter, prevPage, nextPage }) => {
  const [pagi, setPagi] = useState(page);

  useEffect(() => {
    setPagi(page)
  }, [page])

  const handleEnter = useCallback(e => {
    if (e.which === 13) {
      let value = parseInt(e.target.value)
      if (value >= 1 && value <= maxPage) {
        onPressEnter && onPressEnter(value);
      } else if (value > maxPage) {
        onPressEnter && onPressEnter(maxPage);
      } else if (value < 1) {
        onPressEnter && onPressEnter(1);
      }
    }
  });

  const handlePrevPage = useCallback(() => {
    if (page > 1) {
      prevPage && prevPage(page - 1);
    }
  });

  const handleNextPage = useCallback(() => {
    if (page < maxPage) {
      nextPage && nextPage(page + 1);
    }
  });

  return (
    <React.Fragment>
      <div className="pagination">
        {page > 1 && (
          <ButtonClick
            classes="prev"
            onClick={() => handlePrevPage()}
            title="Trang trước"
          />
        )}
        <div className="pageInfo">
          <span>Trang </span>
          <input
            value={pagi}
            onChange={e => setPagi(e.target.value)}
            type="number"
            min="1"
            max={maxPage}
            onKeyPress={e => handleEnter(e)}
            placeholder={page}
          />{" "}
          / {maxPage}
        </div>
        {page < maxPage && (
          <ButtonClick
            classes="next"
            onClick={() => handleNextPage()}
            title="Trang sau"
          />
        )}
      </div>
    </React.Fragment>
  );
};

export default Pagination;
