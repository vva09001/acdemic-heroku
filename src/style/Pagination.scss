$primaryColor: #203e7d;
$secondColor: #0060ae;

.pagination {
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  font-size: 14px;
  font-weight: bold;
  max-height: 35px;
  .pageInfo {
    border-bottom: 2px solid $primaryColor;
    padding: 5px 0;
    margin: 0 10px;
    input {
      background: #dcdcdc;
      max-height: 25px;
      max-width: 70px;
      box-shadow: none;
    }
  }
  button {
    font-size: 12px;
    max-height: 40px;
    color: #fff;
    background: $primaryColor;
    outline-style: none;
    &:hover {
      background: $secondColor;
    }
  }
}

