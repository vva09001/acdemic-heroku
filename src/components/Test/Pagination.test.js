import React from "react";
import { create } from "react-test-renderer";
import Pagination from "../Pagination";

describe("Pagination Component", () => {
  it("Test Pagination component it work", () => {
    const component = create(<Pagination />).toJSON();
    expect(component).toMatchSnapshot();
  });
  exports[`Rerender Pagination component`] = (
    <Pagination
      maxPage={[Number]}
      page={[Number]}
      onPressEnter={[Function]}
      prevPage={[Number]}
      nextPage={[Number]}
    />
  );
});
