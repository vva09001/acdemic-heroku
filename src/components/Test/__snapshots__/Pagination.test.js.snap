// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Test Pagination component it work 1`] = `
<div
  className="pagination"
>
  <div
    className="pageInfo"
  >
    <span>
      Trang 
    </span>
    <input
      min="1"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="number"
    />
     
    / 
  </div>
</div>
`;

exports[`Pagination Component Test Search component it work 1`] = `
<div
  className="pagination"
>
  <div
    className="pageInfo"
  >
    <span>
      Trang 
    </span>
    <input
      min="1"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="number"
    />
     
    / 
  </div>
</div>
`;
